extensions/smallrye-reactive-messaging/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/deployment/SmallRyeReactiveMessagingProcessor.java

```diff
@@ -78,17 +83,26 @@ public void validate(ValidationContext validationContext) {
                 for (BeanInfo bean : validationContext.get(Key.BEANS)) {
                     if (bean.isClassBean()) {
                         // TODO: add support for inherited business methods
-                        for (MethodInfo method : bean.getTarget()
-                                .get()
-                                .asClass()
-                                .methods()) {
+                        ClassInfo ci = bean.getTarget()
+                                .orElseThrow(() -> new IllegalStateException("Target expected")).asClass();
+                        for (MethodInfo method : ci.methods()) {
                             if (annotationStore.hasAnnotation(method, NAME_INCOMING)
                                     || annotationStore.hasAnnotation(method, NAME_OUTGOING)) {
                                 // TODO: validate method params and return type?
                                 mediatorMethods.produce(new MediatorBuildItem(bean, method));
                                 LOGGER.debugf("Found mediator business method %s declared on %s", method, bean);
                             }
                         }
+
+                        for (FieldInfo field : ci.fields()) {
```

Comment:
Hm, this means that we only support field injection. In fact, we can simply iterate over all injection points:
```java
for (InjectionPointInfo injectionPoint : validationContext.get(Key.INJECTION_POINTS)) {
   // TODO
}
```
I'll try to update this PR in a moment...